taccsite_cms/contrib/helpers.py

```diff
@@ -0,0 +1,68 @@
+# Get Django `models.CharField` `choices`
+def get_choices(choice_dict):
+    """Get a sequence for a Django model field choices from a dictionary.
+    :param Dict[str, Dict[str, str]] dictionary: choice as key for dictionary of classnames and descriptions
+    :return: a sequence for django.db.models.CharField.choices
+    :rtype: List[Tuple[str, str], ...]
+    """
+    choices = []
+
+    for key, data in choice_dict.items():
+        choice = (key, data['description'])
+        choices.append(choice)
+
+    return choices
+
+
+
+# GH-93, GH-142, GH-133: Upcoming functions here (ease merge conflict, maybe)
+
+
+
+# Concatenate a list of CSS classes
+# SEE: https://github.com/django-cms/djangocms-bootstrap4/blob/master/djangocms_bootstrap4/helpers.py
+def concat_classnames(classes):
+    """Concatenate a list of classname strings (without failing on None)"""
+    # SEE: https://stackoverflow.com/a/20271297/11817077
+    return ' '.join(_class for _class in classes if _class)
+
+
+
+# GH-93, GH-142, GH-133: Upcoming functions here (ease merge conflict, maybe)
+
+
+
+# Allow plugins to set max number of nested children
+
+from django.shortcuts import render
+
+# SEE: https://github.com/django-cms/django-cms/issues/5102#issuecomment-597150141
+class AbstractMaxChildrenPlugin():
+    """
+    Abstract extension of `CMSPluginBase` that allows setting maximum amount of nested/child plugins.
+    Usage:
+    1. Extend this class,
+       after extending `CMSPluginBase` or a class that extends `CMSPluginBase`.
+    2. Set `max_children` to desired limit.
+    """
+
+    max_children = None
+
+    def add_view(self,request, form_url='', extra_context=None):
+
+        if self.max_children:
+            # FAQ: Placeholders do not have a parent, only plugins do
+            if self._cms_initial_attributes['parent']:
+                num_allowed = len([v for v in self._cms_initial_attributes['parent'].get_children() if v.get_plugin_instance()[0] is not None])
```

Comment:
This is a minor style thing (I also see that it's from a SO answer).
But, I think some would prefer not having list comprehension in long logic like this for readability's sake.
But, again, it is minor.

Reply:
Yes, I got it online. Were I Python-savvy enough to grok this, I would rewrite it with a loop. I will consider improving its legibility, next week. Help is welcome.